src/transformations/horner.cpp

```diff
@@ -343,40 +408,89 @@ polynomial evaluation engine.
     sz    =  2*transformation->order + 2;
     range =  transformation->range;
 
+    const bool iterative_inverse = direction == PJ_INV && transformation->has_only_fwd;
+
     if (direction==PJ_FWD) {                              /* forward */
-        cb =  transformation->fwd_c;
-        c  =  cb + sz;
         e  =  position.u - transformation->fwd_origin->u;
         n  =  position.v - transformation->fwd_origin->v;
         if (transformation->uneg)
             e  =  -e;
         if (transformation->vneg)
             n  =  -n;
     } else {                                              /* inverse */
-        cb =  transformation->inv_c;
-        c  =  cb + sz;
-        e  =  position.u - transformation->inv_origin->u;
-        n  =  position.v - transformation->inv_origin->v;
-        if (transformation->uneg)
-            e  =  -e;
-        if (transformation->vneg)
-            n  =  -n;
+        if (!iterative_inverse) {
+            e  =  position.u - transformation->inv_origin->u;
+            n  =  position.v - transformation->inv_origin->v;
+            if (transformation->uneg)
+                e  =  -e;
+            if (transformation->vneg)
+                n  =  -n;
+        } else {
+            // in this case fwd_origin and any existing flipping needs to be added in the end
+            e = position.u;
+            n = position.v;
+        }
     }
 
     if ((fabs(n) > range) || (fabs(e) > range)) {
         proj_errno_set(P, PROJ_ERR_COORD_TRANSFM_OUTSIDE_PROJECTION_DOMAIN);
         return uv_error;
     }
 
+    if (iterative_inverse) {
+        const double tol = transformation->inverse_tolerance;
+        const std::complex<double> dZ(n-transformation->fwd_c[0], e-transformation->fwd_c[1]);
+        std::complex<double> w0(0.0, 0.0);
+        int loops = 32; // usually converges really fast (1-2 loops)
+        bool converged = false;
+        while (loops-- > 0 && !converged) {
+            // coefficient pointers from back to front until the first complex pair (fwd_c0+i*fwd_c1)
+            double *cb = transformation->fwd_c;
+            double *c = cb + sz;
+            cb += 2;
+            double E = *--c;
+            double N = *--c;
+            while (c > cb) {
+                double w = w0.real()*E + w0.imag()*N + *--c;
+                       N = w0.real()*N - w0.imag()*E + *--c;
+                       E = w;
+            }
+            std::complex<double> det(N, E);
+            std::complex<double> w1 = dZ / det;
+            converged = (fabs(w1.real()-w0.real()) < tol) && (fabs(w1.imag()-w0.imag()) < tol);
```

Comment:
You probably could use `hypot()` or `(w1 - w0).abs()` here?

Reply:
But yes, you are right. So when hypot <= tol, both real and imag will be equal or  less than tol.

Reply:
I agree that it makes strict mathematical sense, and that is probably what matters most for communication purposes. Geodetically, however, the differences in scaling factor absorbed by the polynomals, is related to the historical problem of measuring distances accurately, so the convergence is basically two-dimensional. But as indicated above, I agree that for the iterative convergence the "twice linear" criterion is also reasonable.